### DISCUSSION TITLE
Integration with ExternalProject

### DISCUSSION THREAD

**Discussion by mralhmn** [alandefreitas/matplotplusplus#95] [Q&A]
Hi everybody,

first of all thank you for this awesome project!

I'm new to the CMake world and would like to integrate matplotplusplus with [ExternalProject ](https://cmake.org/cmake/help/v3.10/module/ExternalProject.html) (resp. ExternalProject_Add). My goal is that the user of my program only has to run CMake and start my program and does _not_ have to install matplotplusplus/ download the binary packages manually.

So far I have tried the following:

```
set(MATPLOT_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/matplot")
ExternalProject_Add(
        MATPLOT

        GIT_REPOSITORY https://github.com/alandefreitas/matplotplusplus.git
        GIT_TAG v1.0.1
)

add_dependencies(myproject MATPLOT)
target_link_libraries(myproject matplot)
```

When building my project, JPEG, TIFF, PNG, sgemm_ and so on aren't found (see picture below): 

![grafik](https://user-images.githubusercontent.com/73595847/97444996-0dc26b00-192d-11eb-8aed-f9e988761693.png)

Does anyone know what I'm doing wrong? 

Thanks for every kind of help!

Maria

**Comment by alandefreitas** [comment]
This might solve your first problem: https://alandefreitas.github.io/matplotplusplus/integration/cmake-targets/cmake-with-automatic-download.html

Your second problem is not related to  JPEG, TIFF, PNG, ... It's related to the filesystem library. Your C++ compiler is too old. See instructions for updating here: https://alandefreitas.github.io/matplotplusplus/integration/build-from-source/dependencies.html
